code/datums/components/swabbing.dm

```diff
@@ -0,0 +1,127 @@
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/handle_overlays)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON, .proc/handle_icon)
+
+	src.CanSwabObj = CanSwabObj
+	src.CanSwabTurf = CanSwabTurf
+	src.CanSwabMob = CanSwabMob
+	src.UpdateIcons = UpdateIcons
+	src.UpdateOverlays = UpdateOverlays
+
+/datum/component/swabbing/Destroy()
+	. = ..()
+	for(var/i in swabbed_items)
+		var/datum/D = i
+		qdel(D)
+
+
+///Changes examine based on your sample
+/datum/component/swabbing/proc/examine(datum/source, mob/user, list/examine_list)
+	if(LAZYLEN(swabbed_items))
+		examine_list += "<span class='nicegreen'>There is a microbiological sample on [parent]!</span>"
+		examine_list += "<span class='notice'>You can see the following micro-organisms:</span>\n"
+		for(var/i in swabbed_items)
+			var/datum/biological_sample/samp = i
+			examine_list += samp.GetAllDetails(user.research_scanner) //Get just the names nicely parsed.
+
+///Ran when you attack an object, tries to get a swab of the object. if a swabbable surface is found it will run behavior and hopefully
+/datum/component/swabbing/proc/TryToSwab(datum/source, atom/target, mob/user, params)
+	set waitfor = FALSE //This prevents do_after() from making this proc not return it's value.
+
+	if(istype(target, /obj/structure/table))//help how do i do this less shitty
+		return NONE //idk bro pls send help
+
+	if(istype(target, /obj/item/petri_dish))
+		if(!LAZYLEN(swabbed_items))
+			return NONE
+		var/obj/item/petri_dish/dish = target
+		if(dish.sample)
+			return
+
+		var/datum/biological_sample/deposited_sample
+
+		for(var/i in swabbed_items) //Typed in case there is a non sample on the swabbing tool because someone was fucking with swabbable element
+			if(!istype(i, /datum/biological_sample/))
+				stack_trace("Non biological sample being swabbed, no bueno.")
+				continue
+			var/datum/biological_sample/sample = i
+			//Collapse the samples into one sample; one gooey mess essentialy.
+			if(!deposited_sample)
+				deposited_sample = sample
+			else
+				deposited_sample.Merge(sample)
+
+		dish.deposit_sample(user, deposited_sample)
+		LAZYCLEARLIST(swabbed_items)
+
+		var/obj/item/I = parent
+		I.update_icon()
+
+		return COMPONENT_NO_ATTACK
+	if(!can_swab(target))
+		return NONE //Just do the normal attack.
+
+
+	. = COMPONENT_NO_ATTACK //Point of no return. No more attacking after this.
+
+	if(LAZYLEN(swabbed_items))
+		to_chat(user, "<span class='warning'>You cannot collect another sample on [parent]!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start swabbing [target] for samples!</span>")
+	if(!do_after(user, 30, TRUE, target)) // Start swabbing boi
+		return
+
+	LAZYINITLIST(swabbed_items) //If it isn't initialized, initialize it. As we need to pass it by reference
+
+	if(!SEND_SIGNAL(target, COMSIG_SWAB_FOR_SAMPLES, swabbed_items)) //If we found something to swab now we let the swabbed thing handle what it would do, we just sit back and relax now.
```

Comment:
```suggestion
	if(SEND_SIGNAL(target, COMSIG_SWAB_FOR_SAMPLES, swabbed_items) == NONE) //If we found something to swab now we let the swabbed thing handle what it would do, we just sit back and relax now.
```
Just to be clear that signals return flags, they will never return `null`

Reply:
🤷 thats the same thing

Reply:
Yep, but clarity